### PR TITLE
Add published field to ClassifiedListing Search

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Optimization

## Description
@mstruve [mentioned](https://github.com/thepracticaldev/dev.to/pull/6372#pullrequestreview-367219294) that we need to filter on `published` for `ClassifiedListings` so that, by default, we're only returning `published` `ClassifiedListings`.

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/pull/6372#pullrequestreview-367219294
https://github.com/thepracticaldev/dev.to/projects/6#card-33950122

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## Are there any post deployment tasks we need to perform?
I'll coordinate with @mstruve to manually reindex `ClassifiedListings`.

need_coffee_gif

### CODE REVIEW

File: app/services/search/query_builders/classified_listing.rb [hunk 1]
```diff
@@ -30,6 +32,7 @@ class ClassifiedListing
 
       def initialize(params)
         @params = params.deep_symbolize_keys
+        @params[:published] = DEFAULT_PARAMS[:published] unless @params.key?(:published)
```
Comment (mstruve) [inline app/services/search/query_builders/classified_listing.rb]: We can keep this `@params.key?(:published)` here as long as we don't permit `published` as a param for the controller. I don't want people having the ability to override this from the frontend.

Comment (atsmith813) [reply app/services/search/query_builders/classified_listing.rb]: Ah, I see. I thought we might need that functionality in the future for the API or something. I'll just remove this altogether so it's always `true`. And if we need it later, we can always add it in at that time. How does that sound?

Comment (mstruve) [reply app/services/search/query_builders/classified_listing.rb]: Sounds good

**Comment by rhymes** [comment]
I just pulled the code locally and thought that maybe we should have added a data update script to reindex `ClassifiedListing` rows here since we needed to. My bad I didn't think about this sooner while reviewing the code :-/ 

cc @mstruve

**Comment by atsmith813** [comment]
That's a good point about needing that locally. I had discussed reindexing before this PR and we decided to do it manually in the console since a DataScript to index ClassifiedListings exists already.

A quick and dirty way to do it locally is to delete the existing script locally on your machine and it will re-run it. Or perform it locally in the console for a quick fix 👍

**Comment by mstruve** [comment]
@rhymes you can blame me for this, I figured it was a quick reindex and since the indexing PR hasn't been out that long many would not have even run the DataUpdateScript yet so I figured we could sneak it in.

**Comment by rhymes** [comment]
Ahah no blame, was just wondering, maybe we did forget :D I'm going to re-index locally then, thanks!

**Pull request merged**